test/integration/idle_timeout_integration_test.cc

```diff
@@ -50,21 +55,25 @@ class IdleTimeoutIntegrationTest : public HttpProtocolIntegrationTest {
 
   void sleep() { test_time_.timeSystem().sleep(std::chrono::milliseconds(TimeoutMs / 2)); }
 
-  void waitForTimeout(IntegrationStreamDecoder& response) {
+  void waitForTimeout(IntegrationStreamDecoder& response, absl::string_view stat_name = "",
+                      absl::string_view stat_prefix = "http.config_test") {
     if (downstream_protocol_ == Http::CodecClient::Type::HTTP1) {
       codec_client_->waitForDisconnect();
     } else {
       response.waitForReset();
       codec_client_->close();
     }
-    EXPECT_EQ(1, test_server_->counter("http.config_test.downstream_rq_idle_timeout")->value());
+    if (!stat_name.empty()) {
+      EXPECT_EQ(1, test_server_->counter(fmt::format("{}.{}", stat_prefix, stat_name))->value());
+    }
   }
 
   // TODO(htuch): This might require scaling for TSAN/ASAN/Valgrind/etc. Bump if
   // this is the cause of flakes.
   static constexpr uint64_t TimeoutMs = 200;
   bool enable_global_idle_timeout_{};
   bool enable_per_stream_idle_timeout_{true};
+  bool enable_request_timeout_{};
```

Comment:
"You didn't cause the problem but can you fix it anyway?": I don't like that some of these bools are defaulted (presumably false) and some specified; it's inconsistent.  Could you switch from {} to {false} for enable_global_idle_timeout_ and enable_request_timeout_?